### PR TITLE
Add javascript bindings for calib3d module.

### LINKED ISSUE #14293: Webassembly build does not contain the calib3d module
##### System information (version)

- OpenCV => 4.1.0
- Operating System / Platform => Ubuntu 18.10
- Compiler => Emscripten

##### Detailed description

I am wanting to use the [`findHomography`](https://docs.opencv.org/2.4/modules/calib3d/doc/camera_calibration_and_3d_reconstruction.html#findhomography) function in the webassembly build of opencv. However, no bindings are provided for the calib3d module. By the [comment in the build script](https://github.com/opencv/opencv/blob/64168fc20aa8a914cb5529f90ffac309854563b1/platforms/js/build_js.py#L119) this is intentional.

However, I wanted to start a discussion on whether it is feasible to include this module, and if so what would be needed to be done to get this working.

### PR DESCRIPTION
**Pull request by cameron-martin**
resolves #14293

### This pullrequest changes

This adds javascript bindings for the calib3d module, along with tests.

Currently I have only added the `findHomography` function, but will add the rest of the module soon.

<cut/>

```
force_builders_only=Docs,Custom
build_image:Docs=docs-js
build_image:Custom=javascript
buildworker:Custom=linux-4
```

### CODE REVIEW

File: modules/js/test/test_calib3d.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
+// //////////////////////////////////////////////////////////////////////////////////////
```
Comment (dkurt) [inline modules/js/test/test_calib3d.js]: Please replace to the shorter version of header:

https://github.com/cameron-martin/opencv/blob/41018e0cb7375ccf87bd0af0df3e6755812fea78/modules/js/test/test_features2d.js

File: modules/js/src/helpers.js [hunk 2]
```diff
@@ -397,3 +397,6 @@ Module['matFromImageData'] = function(imageData) {
     mat.data.set(imageData.data);
     return mat;
 };
+
+Module['CV_RANSAC'] = 8;
+Module['CV_LMEDS'] = 4;
```
[the remark this answers is not in the document]
Comment (cameron-martin) [reply modules/js/src/helpers.js]: Ah, I was looking at an old version of the docs. Fixed.

File: modules/js/test/test_calib3d.js (hunk 1 above)
Comment (cameron-martin) [reply modules/js/test/test_calib3d.js]: Fixed.

**Comment by dkurt** [comment]
Hi! I rebased this PR with the latest master branch.

**Pull request merged**